widgetssdk/build.gradle

```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 31
+    compileSdkVersion 33
     buildToolsVersion "30.0.3"
     defaultPublishConfig "debug"
     defaultConfig {
         minSdkVersion 24
-        targetSdkVersion 31
+        targetSdkVersion 33
```

Comment:
I remember one of our integrators faced with some issues when we upgraded `targetSdkVersion`. @gugalo was working on that request. But I don't remember the details.
@gugalo, could you please remind me what was the issue and if `targetSdkVersion` won't cause any issues for our integrators?

Reply:
It shouldn't be an issue.

Reply:
It is not an issue on its own however I still would do that especially since you have caused the same issue as last time the integrator complained about this.

If we just upgrade the `compileSdkVersion` it is a good thing and won't cause any issues to the integrators. But the problem comes because of easy-to-miss changes in dependencies versions. Android dependencies now can require a minimum `compileSdkVersion` using a `minCompileSdk` property. And you just introduced in your changes `androidx.core:core-ktx:1.9.0` that requires compileSdkVersion 33+.

For example, this is an error that the integrator would get if they use Widgets from your PR with compiled SDK of a lower version:
```
One or more issues found when checking AAR metadata values:

Dependency 'androidx.core:core:1.9.0' requires 'compileSdkVersion' to be set to 33 or higher.
Compilation target for module ':app' is 'android-31'

Dependency 'androidx.core:core-ktx:1.9.0' requires 'compileSdkVersion' to be set to 33 or higher.
Compilation target for module ':app' is 'android-31'

Dependency 'androidx.annotation:annotation-experimental:1.3.0' requires 'compileSdkVersion' to be set to 33 or higher.
Compilation target for module ':app' is 'android-31'
```

You could argue that integrators can manually force override dependencies version in their build.gradle. And that is exactly what I shared with the client last time. But that requires additional effort and I don't remember why they had problems doing that even after I shared clear interactions with them. 

For reference here is the link to the old issue 
https://glia.atlassian.net/browse/MOB-1235
Slack discussion: https://salemove.slack.com/archives/CHN6UB7UH/p1650279001120289

Reply:
We might want to discuss and see how we handle this in the future. @AndriiHorishniiMOC @andrews-moc @DavDo @mikktammeglia.

Reply:
For now, though the simplest solution would be to downgrade the `androidx.core` to 1.8.0 which has `minCompileSdk=31`.
Same for @DavDo PR

Reply:
@gugalo Thank you for the explanation. According to the Glia Android Core SDK, we should decrease the value to 32.
https://github.com/salemove/android-sdk/blob/master/androidSdk/build.gradle#L11

Reply:
That's actually strange because we should not have a higher compile SDK the target SDK requirements for Google Play Store upload which currently is API 31 until November 1, 2023 when it becomes API 33 (skipping the 32 version).

But that's ideally not sure if any of this is a real issue so I think we might just make some agreement that we as a team thinks is reasonable and just follow whatever we agreed upon.

Reply:
Might also incode someone related to the client engeneering in the discussion.

Reply:
@mikktammeglia can I ask you to organize the meeting since I am not sure who are the parties that should be invited to the discussion

Reply:
Hei,

Discussed it a bit and we can have a short round of discussion after start-up tomorrow. If I understand things correctly:
1. It is easier for our clients if we use the minimum. (31 for now).
2. We can change this policy if there are specific gains we can say from having a higher version. 
3. If it makes sense then tomorrow in the engineering initiatives discussion it could be added as one more initiative to make it work with the highest one. (i understand we could add some checks to it)

But if I misunderstood anything I am happy to be shown the value of going with highest in as precise way as possible. :)

Reply:
@gugalo Sorry for the misunderstanding. The Core SDK has `targetSdkVersion` 32, not `minSdkVersion`.